src/filters.rs

```diff
@@ -244,6 +244,30 @@ pub fn last(input: &Value, _args: &[Value]) -> FilterResult {
     }
 }
 
+pub fn split(input: &Value, args: &[Value]) -> FilterResult {
+    // Make sure there is only 1 argument to split
+    if args.len() != 1 {
+        return Err(InvalidArgumentCount(format!("expected 1, {} given", args.len())));
+    }
+
+
+    match *input {
+        Str(ref string_to_split) => {
+            // the input String is in fact a String
+            match args.first() { // Check the first (and only) argument
+                Some(&Str(ref split_string)) => {
+                    // The split string argument is also in fact a String
+                    let v: Vec<&str> = string_to_split.split(split_string).collect();
+                    // Split and construct result
+                    Ok(Array(v.iter().map(|x| Str(String::from(*x))).collect()))
```

Comment:
you're calling `.collect()` and immediately afterwards `.iter()`. I think you can simply remove both these calls.